### ISSUE TITLE
Can't resolve 'fs', Can't resolve 'path', Can't resolve 'stream', Cannot find name 'require'

### ISSUE THREAD

**Issue by isimonxd** [gitbrent/PptxGenJS#366]
I am trying to run PptxGenJS (version 2.2.0.20180617 (from using pptx.version)) on Angular 6 but am getting a bunch of errors:
```
WARNING in ./node_modules/pptxgenjs/dist/pptxgen.js
Module not found: Error: Can't resolve 'fs' in 'C:\Users\Project\pptgen\node_modules\pptxgenjs\dist'
ERROR in ./node_modules/image-size/lib/index.js
Module not found: Error: Can't resolve 'fs' in 'C:\Users\Project\pptgen\node_modules\image-size\lib'
ERROR in ./node_modules/image-size/lib/types/tiff.js
Module not found: Error: Can't resolve 'fs' in 'C:\Users\Project\pptgen\node_modules\image-size\lib\types'
ERROR in ./node_modules/image-size/lib/index.js
Module not found: Error: Can't resolve 'path' in 'C:\Users\Project\pptgen\node_modules\image-size\lib'
ERROR in ./node_modules/jszip/lib/readable-stream-browser.js
Module not found: Error: Can't resolve 'stream' in 'C:\Users\Project\pptgen\node_modules\jszip\lib'
i ｢wdm｣: Failed to compile.
ERROR in src/app/pptgen/pptgen.component.ts(43,23): error TS2304: Cannot find name 'require'.
```
The code that I am using is:
```
const PptxGenJS = require('pptxgenjs');
const  pptx = new PptxGenJS(); 
pptx.setBrowser(true);
const  slide = pptx.addNewSlide();
...
```

However, when running the program without "const PptxGenJS = require('pptxgenjs');" I would get an error stating:
```
ERROR in src/app/pptgen/pptgen.component.ts(44,23): error TS2304: Cannot find name 'PptxGenJS'.
```
The powerpoint generator would sometimes work or give me a "Cannot GET /".
Code:
```
const  pptx = new PptxGenJS(); 
pptx.setBrowser(true);
const  slide = pptx.addNewSlide();
...
```

I'm not too sure where to go from here now to ensure that the pptxgenjs works at all times. Thank you in advance.

**Comment by gitbrent** [comment]
Are you using webpack?  Can you use the common workaround:

Webpack config:
```
node: {
   fs: "empty"
}
```

Angular example:
* [PptxGenJS Library Integration](https://gitbrent.github.io/PptxGenJS/docs/integration.html)

**Comment by isimonxd** [comment]
@gitbrent Thanks for the reply. I'm not using webpack. But when I tried to open the webpack.config.js file with "ng eject" to included "node: { fs: "empty" }", I got an issue that the 'eject' command has been temporarily disabled. Are there any other ways I can do to workaround this issue? Or another way to generate a webpack.config.js file without ng eject.

**Comment by dgabriel** [comment]
Unfortunately, I doesn't look like Angular is going to re-enabled any time soon.  I'm trying to figure out a workaround now.  @isimonxd, did you manage to get this working?

**Comment by dgabriel** [comment]
Yes, it's not possible to use 2.3.0 with angular right now.  I forked the project to see if I can get it to play nice.

**Comment by dgabriel** [comment]
Please see https://github.com/gitbrent/PptxGenJS/issues/407#issuecomment-423614710

**Comment by gitbrent** [comment]
Hopefully fixed with Issue #415 and `package.json` changes coming in Version 2.4.0
